src/mgr/TTLCache.h

```diff
@@ -0,0 +1,159 @@
+#pragma once
+
+#include "PyUtil.h"
+#include <atomic>
+#include <boost/optional.hpp>
+#include <chrono>
+#include <functional>
+#include <map>
+#include <memory>
+#include <string>
+
+using namespace std;
+
+template <class Key, class Value> class Cache {
+private:
+  std::atomic<uint64_t> hits, misses;
+
+protected:
+  unsigned int capacity;
+  Cache(unsigned int size = UINT16_MAX) : hits{0}, misses{0}, capacity{size} {};
+  std::map<Key, Value> content;
+
+  void miss() { misses++; }
+  void hit() { hits++; }
+
+  unsigned int get_misses() { return misses; }
+  unsigned int get_hits() { return hits; }
+
+public:
+  void insert(Key key, Value value) {
+    if (content.size() < capacity) {
+      content.insert({key, value});
+    }
+  }
+  Value get(Key key) { return content[key]; }
+  void erase(Key key) { content.erase(content.find(key)); }
+  void clear() { content.clear(); }
+  bool exists(Key key) { return content.find(key) != content.end(); }
```

Comment:
When available, rather than actively checking, why not going for the 'ask forgiveness than to permission' motto?
```c++
   try {
     return content.at(key);
   } catch (const std::out_of_range& oor) {
     return std::nullopt;
   }
```